libfwupdplugin/fu-device.c

```diff
@@ -218,6 +222,12 @@ fu_device_add_internal_flag (FuDevice *self, FuDeviceInternalFlags flag)
 	FuDevicePrivate *priv = GET_PRIVATE (self);
 	g_return_if_fail (FU_IS_DEVICE (self));
 	priv->internal_flags |= flag;
+
+	/* backwards compat */
+	if (flag & FU_DEVICE_INTERNAL_FLAG_REPLUG_MATCH_GUID)
+		fu_device_add_flag (self, FWUPD_DEVICE_FLAG_NO_GUID_MATCHING);
```

Comment:
is this really the right match to add to the device side?  It seems like an opposite behavior.

Reply:
my original idea was to force the plugin to either opt-in with FU_DEVICE_INTERNAL_FLAG_REPLUG_MATCH_GUID or opt-out with FWUPD_DEVICE_FLAG_NO_GUID_MATCHING -- although now I'm checking all three before the g_warning. I agree it's confusing. I think we can just drop these.

Reply:
Removed.